lib/housekeeping.js

```diff
@@ -142,35 +146,38 @@ class HouseKeeper {
         totalGbUsage += combinedGbUsage;
         totalCount += combinedCount;
         
-        let transaction = await this.state.beginTransaction();
-        try {
-          await this.state.reportEbsUsage({
-            region,
-            volumetype: type,
-            state: 'active',
-            totalcount: active.count,
-            totalgb: active.gb,
-          }, transaction);
-          
-          await this.state.reportEbsUsage({
-            region,
-            volumetype: type,
-            state: 'unused',
-            totalcount: unused.count,
-            totalgb: unused.gb,
-          }, transaction);
-          
-          await this.state.commitTransaction(transaction);
-        } catch (err) {
-          await this.state.rollbackTransaction(transaction);
-          log.error(err, 'Error reporting EBS usage');
-        }
+        rowData.push({
+          region, 
+          volumetype: type,
+          state: 'active', 
+          totalcount: active.count,
+          totalgb: active.gb,
+        });
+        rowData.push({
+          region, 
+          volumetype: type,
+          state: 'unused', 
+          totalcount: unused.count,
+          totalgb: unused.gb,
+        });
       }));
     }));
   
     // Update global totals after all regions have been processed
     volumesMonitor.measure('global.gb-usage', totalGbUsage);
     volumesMonitor.measure('global.count', totalCount);
+
+    if (rowData.length > 0) {
+      // Submit row data to ebsusage table
+      let transaction = await this.state.beginTransaction();
+      try {
+        await this.state.reportEbsUsage(rowData); //, transaction);
+        await this.state.commitTransaction(transaction);
+      } catch (err) {
+        await this.state.rollbackTransaction(transaction);
+        log.error(err, 'Error reporting EBS usage');
+      }
+    }
```

Comment:
It looks like this wraps the `reportEbsUsage` call in a transaction, but `reportEbsUsage` also uses its own transaction.  In fact since the `transaction` arg is commented out here, I think this transaction isn't used at all and could be removed (making this a one-line call to `reportEbsUsage`).

Reply:
Whoa, this is interesting. None of the tests fail for me locally :confused: